lib/user.rb

```diff
@@ -0,0 +1,26 @@
+class User
+  attr_reader :name, :jokes
+
+  def initialize(name)
+    @name = name
+    @jokes = []
+  end
+
+  def learn(joke)
+    @jokes << joke
+  end
+
+  def jokes
+    @jokes
+  end
+
+  def tell(user, joke)
+    @jokes << joke
+  end
+
+  def joke_by_id(id)
+    @jokes.find_all do |joke|
+      joke.id == id
+    end
+  end
```

Comment:
Nice work here. We can simplify it since each joke should have a unique ID value, and use `find` instead of `find_all`, which will stop as soon as it finds a match. `find_all` will continue to look through the rest of the jokes to collect everything that matches.